### PR TITLE
Profile Editing

### PR DESCRIPTION
**Pull request by sdub18**
One of the first things I wanted to work on since downloading Damus. Everyone needs a profile picture right?

Draft for now. The logic is there but profile editing has been inconsistent. I haven't been able to consistently get my profile to  update with new details, however I have had it work a few times and on different accounts.

This PR also finally makes the `DamusViewModel` class I have been dying to add in. You can see it working as intended on the EditAccountView where it's injected as an environment object.

The UI is there (feel free to make suggestions where you see fit) and the logic is there but definitely needs more work before it can go into production.

Let me know what you guys think!

Screenshot 2022-12-21 at 1 02 14 PM

### CODE REVIEW

**Comment by jb55** [comment]
thank god

**Comment by jb55** [comment]
this is good for now but I do want an option to upload which we can add later.

**Comment by sdub18** [comment]
Yeah absolutely agree

**Comment by sdub18** [comment]
> thank god
😂

**Review by kraliq0 — APPROVED** [review]

File: damus.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8,6 +8,9 @@
 
 /* Begin PBXBuildFile section */
 		3169CAE6294E69C000EE4006 /* EmptyTimelineView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3169CAE5294E69C000EE4006 /* EmptyTimelineView.swift */; };
+		31D2E84D295225C3006D67F8 /* EditAccountView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 31D2E84C295225C3006D67F8 /* EditAccountView.swift */; };
```
Comment (kraliq0) [inline damus.xcodeproj/project.pbxproj:11]: Molecular Soul

**Review by kraliq0 — APPROVED** [review]

**Comment by jb55** [comment]
lots of issues with this relay/logic wise. shall I take a pass at it?

**Comment by sdub18** [comment]
Yes that would be so helpful. I was reading up on the docs to try and understand the protocol more but I would really appreciate a helping hand.

I was basing it off of the nips as well as how you wrote the createAccount metaRequest. Would be helpful in learning more on how to do it in the future.

**Comment by jb55** [comment]
I ended up using another PR for this, but feel free to tweak the EditMetadataView, buttons, etc now to your liking!

**Pull request closed without merging**